src/test/java/io/reactivex/internal/operators/single/SingleTimeoutTest.java

```diff
@@ -69,4 +70,133 @@ public void mainError() {
         .awaitDone(5, TimeUnit.SECONDS)
         .assertFailure(TestException.class);
     }
+
+    @Test
+    public void disposeWhenFallback() {
+        TestScheduler sch = new TestScheduler();
+
+        SingleSubject<Integer> subj = SingleSubject.create();
+
+        subj.timeout(1, TimeUnit.SECONDS, sch, Single.just(1))
+        .test(true)
+        .assertEmpty();
+
+        assertFalse(subj.hasObservers());
+    }
+
+    @Test
+    public void isDisposed() {
+        TestHelper.checkDisposed(SingleSubject.create().timeout(1, TimeUnit.DAYS));
+    }
+
+    @Test
+    public void fallbackDispose() {
+        TestScheduler sch = new TestScheduler();
+
+        SingleSubject<Integer> subj = SingleSubject.create();
+
+        SingleSubject<Integer> fallback = SingleSubject.create();
+
+        TestObserver<Integer> to = subj.timeout(1, TimeUnit.SECONDS, sch, fallback)
+        .test();
+
+        assertFalse(fallback.hasObservers());
+
+        sch.advanceTimeBy(1, TimeUnit.SECONDS);
+
+        assertFalse(subj.hasObservers());
+        assertTrue(fallback.hasObservers());
+
+        to.cancel();
+
+        assertFalse(fallback.hasObservers());
+    }
+
+    @Test
+    public void normalSuccessDoesntDisposeMain() {
+        final int[] calls = { 0 };
```

Comment:
nit: I know you often use arrays for that, but `AtomicInteger` kinda much more readable when you need a reference to single "mutable" int :)